### ISSUE TITLE
[Feature]: Support Native JSON columns in Clickhouse

### ISSUE THREAD

**Issue by navinpai** [jaegertracing/jaeger-clickhouse#112]
### Requirement

As a Clickhouse analytics user, I want the `clickhouse-jaeger` schema to allow using Clickhouse native JSON columns so that we can query data in clickhouse more efficiently ([both in terms of performance and query simplicity](https://presentations.clickhouse.com/release_22.3/#20))

### Problem

Currently, Clickhouse-Jaeger stores JSON span data as a [string column-type](https://github.com/jaegertracing/jaeger-clickhouse/blob/main/sqlscripts/jaeger-spans.tmpl.sql#L9), which makes it quite verbose to have to query based on fields within the column using [Clickhouse's JSON functions](https://clickhouse.com/docs/en/sql-reference/functions/json-functions/) , especially if you get past 2 levels of nesting.

This is very evident, when you want to query the ingested data to generate your own analytics/insights. It would be nice if jaeear-clickhouse added support for Clickhouse native JSON columns

### Proposal

A solution may be to start providing support for the native JSON datatype _(It's still "[experimental](https://clickhouse.com/docs/en/sql-reference/data-types/json/)", but the spec has been quite stable for a while)_

### Open questions

The major open question is how this would affect the split between protobuf and json encoded data _(currently, string supports both)_ and whether it'll add more complexities to the project. Need to observe more to see the impact of this, but wanted to raise this with the community/maintainers to get an idea of their thoughts.

**Comment by chhetripradeep** [comment]
@navinpai I agree from read perspective, reading a field from a JSON datatype is faster than reading a field from String datatype but from write perspective, inserting JSON columns are more expensive & hence slower than inserting String columns. And generally in metrics/logs/tracing system, we do far more writes than reads, so i feel string datatype is more appropriate. Feel free to correct me if my understanding is wrong.
